core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -225,6 +248,31 @@ public CompiledScript compile(Script script, ScriptContext scriptContext, Map<St
     }
 
     /**
+     * Check whether there have been too many compilations within the last minute, throwing a circuit breaking exception if so.
+     */
+    public void checkCompilationLimit() {
```

Comment:
If this method took a `Supplier<Long>` then you could test time passing in the test.

Also, should it be package private?